src/documents/forms/EditDocumentForm.js

```diff
@@ -141,12 +161,11 @@ const EditDocumentForm = React.forwardRef(
                 </Form.Field>
                 {submitButtons &&
                   submitButtons(
-                    Object.values(values).every(x =>
-                      Boolean(x !== undefined),
-                    ) ||
-                      Object.values(errors.file_name || {}).some(
-                        x => x != null,
-                      ),
+                    Object.keys(errors).length > 0 ||
```

Comment:
same here with the conditional length, maybe we can abstract these up to make them more resuable and readable (again my fault for making them daft to start with )